custom_components/switch_humidifier/humidifier.py

```diff
@@ -205,7 +205,8 @@ def turn_off(self, **kwargs):
 
   def _state_changed(self, entity_id, old_state, new_state):
     """Called on sensor or switch state change"""
-    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable':
+    if not new_state is None and not new_state.state == 'unknown' and not new_state.state == 'unavailable' and old_state.state != new_state.state:
+      # old switch power and new switch power must be different
```

Comment:
This method is for any state change (humidity sensor or switch), not only switch changes.
Can you update the comment to something like "Check if they are different states"